reporting_app/__init__.py

```diff
@@ -574,3 +564,51 @@ def species_page(species):
             )
         ]
     )
+
+
+@app.route('/charts/<view_type>')
+@flask_login.login_required
+def plotting_report(view_type):
+    if view_type == 'last_month':
+        time_ago = util.now() - datetime.timedelta(days=30)
+    elif view_type == 'last_3_months':
+        time_ago = util.now() - datetime.timedelta(days=182)
+    elif view_type == 'last_12_months':
+        time_ago = util.now() - datetime.timedelta(days=365)
+    else:
+        fl.abort(404)
+        return None
+
+    return render_template(
+        'charts.html',
+        api_urls=[
+            util.construct_url('lanes', max_results=10000, where={'_created': {'$gte': time_ago.strftime(settings.DATE_FORMAT)}}),
+            util.construct_url('lims/run_status', createddate=time_ago.strftime(settings.DATE_FORMAT)),
+        ],
+        ajax_token=util.get_token(),
+        merge_on='run_id',
+        merge_properties=['run'],
+        metric_options=chart_metrics_mappings.get('seq_plot_metrics'),
+        color_options=chart_metrics_mappings.get('seq_plot_colors')
+    )
+
+
+@app.route('/tat_chart/')
+@flask_login.login_required
+def tat_chart_page(view_type):
+    time_ago = None
+    if view_type == 'last_month':
+        time_ago = util.now() - datetime.timedelta(days=30)
+    if view_type == 'last_3_months':
```

Comment:
This needs to be an `elif`, otherwise `last_month` hits a 404 Not Found

Reply:
This code has been refactored